vortex-array/src/array/bool/mod.rs

```diff
@@ -83,8 +83,7 @@ impl BoolArray {
     /// otherwise a copy is created.
```

Comment:
There are a bunch of changes in this file that are due to a [change](// let first_byte_bit_offset) in arrow. There's probably a different way to go that maintains our end here and probably makes better shallow clones of the arrow buffer.